openedx/features/enterprise_support/api.py

```diff
@@ -430,6 +431,31 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
     return consent_needed
 
 
+def consent_needed_for_courses(user, course_ids):
+    """
+    For a list of course_ids, this function determines if the user needs to grant
+    data sharing permissions before accessing the course.
+    Note that this function makes use of the Enterprise models directly instead of using the API calls
+    """
+    result = {course_id: False for course_id in course_ids}
+    if not enterprise_enabled():
+        return result
```

Comment:
should do the check for enterprise_enabled() first and don't loop through the courses if it's not enabled. just return None or empty dict in that case.